### PR TITLE
Add mpConfig test with config vars and add archive marker

### PR DESCRIPTION
**Pull request by tjwatson**

### CODE REVIEW

**Comment by tjwatson** [comment]
#build

File: dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java [hunk 1]
```diff
@@ -120,18 +99,15 @@ private void configureBeforeRestore() {
...
+                case varDirValueTest:
+                    server.copyFileToLibertyServerRoot("variables", "configVariables/req_scope_key");
                     break;
                 case noDefaultEnvValueTest:
                     configureEnvVariable(server, singletonMap("optional_req_scope_key", "optionalEnvValue"));
                     break;
                 case noDefaultServerValueTest:
                     updateVariableConfig(server, "optional_req_scope_key", "optionalServerValue");
                     break;
-                case providerEnvValueChangeTest:
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:132]: `providerEnvValueTest` can come here.

Comment (tjwatson) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:132]: Nice catch.  Yes I need to add two provider tests: one for req scope and one for app scope

File: dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java [hunk 2]
```diff
@@ -143,18 +119,15 @@ private void configureBeforeRestore() {
                 case appScopeAnnoValueTest:
                     removeVariableConfig("app_scope_key");
                     break;
-                case appScopeEnvValueChangeTest:
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:146]: `appScopeProviderEnvValueTest` can come here.

File: dev/io.openliberty.checkpoint_fat_mpconfig/test-applications/mpapp1/src/mpapp1/MPConfigServlet.java [hunk 3]
```diff
@@ -76,11 +76,6 @@ public void noDefaultServerValueTest() {
         reqScopeBean.noDefaultServerValueTest();
     }
 
-    @Test
-    public void providerEnvValueChangeTest() {
-        reqScopeBean.providerEnvValueChangeTest();
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/test-applications/mpapp1/src/mpapp1/MPConfigServlet.java:81]: @Test
    public void providerEnvValueTest() {
        reqScopeBean.providerEnvValueTest();
    }

File: dev/io.openliberty.checkpoint_fat_mpconfig/test-applications/mpapp1/src/mpapp1/MPConfigServlet.java [hunk 4]
```diff
@@ -117,11 +112,6 @@ public void appScopeNoDefaultServerValueTest() {
         appScopeBean.appScopeNoDefaultServerValueTest();
     }
 
-    @Test
-    public void appScopeProviderEnvValueChangeTest() {
-        appScopeBean.appScopeProviderEnvValueChangeTest();
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/test-applications/mpapp1/src/mpapp1/MPConfigServlet.java:122]: @Test
    public void appScopeProviderEnvValueTest() {
        appScopeBean.appScopeProviderEnvValueTest();
    }

File: dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java [hunk 5]
```diff
@@ -143,18 +119,15 @@ private void configureBeforeRestore() {
...
+                    server.copyFileToLibertyServerRoot("variables", "configVariables/app_scope_key");
                     break;
                 case appScopeNoDefaultEnvValueTest:
                     configureEnvVariable(server, singletonMap("optional_app_scope_key", "optionalEnvValue"));
                     break;
                 case appScopeNoDefaultServerValueTest:
                     updateVariableConfig(server, "optional_app_scope_key", "optionalServerValue");
                     break;
-                case appScopeProviderEnvValueChangeTest:
-                    configureEnvVariable(server, singletonMap("provider_app_scope_key", "providerEnvValueChange"));
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:156]: case appScopeProviderEnvValueTest:
                    configureEnvVariable(server, singletonMap("provider_app_scope_key", "providerEnvValue"));

File: dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java [hunk 6]
```diff
@@ -120,18 +99,15 @@ private void configureBeforeRestore() {
...
+                    server.copyFileToLibertyServerRoot("variables", "configVariables/req_scope_key");
                     break;
                 case noDefaultEnvValueTest:
                     configureEnvVariable(server, singletonMap("optional_req_scope_key", "optionalEnvValue"));
                     break;
                 case noDefaultServerValueTest:
                     updateVariableConfig(server, "optional_req_scope_key", "optionalServerValue");
                     break;
-                case providerEnvValueChangeTest:
-                    configureEnvVariable(server, singletonMap("provider_req_scope_key", "providerEnvValueChange"));
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:133]:
```
case providerEnvValueTest:
                    configureEnvVariable(server, singletonMap("provider_req_scope_key", "providerEnvVal"));
 ```

File: dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java [hunk 7]
```diff
@@ -120,18 +99,15 @@ private void configureBeforeRestore() {
...
-                    configureEnvVariable(server, singletonMap("req_scope_key", "envValueChange"));
+                case varDirValueTest:
+                    server.copyFileToLibertyServerRoot("variables", "configVariables/req_scope_key");
                     break;
                 case noDefaultEnvValueTest:
                     configureEnvVariable(server, singletonMap("optional_req_scope_key", "optionalEnvValue"));
                     break;
                 case noDefaultServerValueTest:
                     updateVariableConfig(server, "optional_req_scope_key", "optionalServerValue");
                     break;
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:110]: `providerEnvValueTest` can come here.

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#libby

File: dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java [hunk 8]
```diff
@@ -229,43 +202,38 @@ private void checkForLogsAndStopServer() throws Exception {
...
 
                 // MPConfigBeanWithApplicationScope bean
                 case appScopeEnvValueTest:
                 case appScopeServerValueTest:
                 case appScopeAnnoValueTest:
-                case appScopeEnvValueChangeTest:
                 case appScopeNoDefaultEnvValueTest:
                 case appScopeNoDefaultServerValueTest:
-                case appScopeProviderEnvValueChangeTest:
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:246]: we can add `appScopeProviderEnvValueTest` and `appScopeVarDirValueTest` here.

File: dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java [hunk 9]
```diff
@@ -229,43 +202,38 @@ private void checkForLogsAndStopServer() throws Exception {
                 case envValueTest:
                 case serverValueTest:
                 case annoValueTest:
-                case envValueChangeTest:
+                case providerEnvValueTest:
                 case noDefaultEnvValueTest:
                 case noDefaultServerValueTest:
-                case providerEnvValueChangeTest:
```
Comment (anjumfatima90) [inline dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:235]: we can add `varDirValueTest` here.

Comment (tjwatson) [reply dev/io.openliberty.checkpoint_fat_mpconfig/fat/src/io/openliberty/checkpoint/fat/MPConfigTest.java:235]: Done

**Review by anjumfatima90 — APPROVED** [review]

**Comment by tjwatson** [comment]
#build
#libby

**Pull request merged**